### PR TITLE
Codebase upgrade

### PR DESCRIPTION
**Pull request by vorot93**
This is a very large pull request that includes the upgrade to latest dependency versions and fixes suggested by Clippy.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,34 +12,34 @@ readme = "README.md"
...
-ethabi = "7.0.0"
-ethereum-types = "0.5.2"
-futures = "0.1.26"
-jsonrpc-core = "11.0.0"
-log = "0.4.6"
-parking_lot = "0.7.1"
-rustc-hex = "2.0.1"
-serde = { version="1.0.90", features = ["derive"] }
-serde_json = "1.0.39"
+arrayvec = "0.4"
```
Comment (tomusdrw) [inline Cargo.toml]: I'd prefer to keep full versions here. I'm totally fine with having shorter versions in applications/binaries, but not in libraries.
The reason for this is that depending on other libraries cargo might resolve  `0.4` to `0.4.0` and if there are any new features added in say `0.4.1` that we actually use the library won't compile.
So it's better to have a minimal required version encoded here.

Comment (vorot93) [reply Cargo.toml]: My concern was the opposite one: explicitly listing minor versions will inevitably cause dependencies to quickly become stale. By leveraging semver we allow the bugfixes in dependencies to automatically roll into dependency graph while keeping breaking changes out.

Comment (tomusdrw) [reply Cargo.toml]: Sure, but using `="0.4.10"` syntax only specifies the minimal required version, cargo will resolve to any bugfix release out there (i.e. `0.4.11` is fine in Cargo.lock), so I think we are on the same page.

Comment (vorot93) [reply Cargo.toml]: @tomusdrw Seems so. Fixed.

Comment (tomusdrw) [reply Cargo.toml]: Perfect :ok_hand:, thanks again!

**Pull request merged**